pipelines/genomics/docker/run.sh

```diff
@@ -0,0 +1,141 @@
+# for each individual read and its alignment to the genome. While we do
+# not have time to go in detail of the features of the SAM format, the
+# paper by Heng Li et al. provides a lot more detail on the specification.
+# The binary version of SAM is called a BAM file.
+#
+# The file begins with a header, which is optional.
+# The header is used to describe source of data, reference sequence,
+# method of alignment, etc., this will change depending on the aligner
+# being used. Following the header is the alignment section.
+# Each line that follows corresponds to alignment information
+# for a single read. Each alignment line has 11 mandatory fields for
+# essential mapping information and a variable number of other fields
+# for aligner specific information. An example entry from a SAM file is
+# displayed below with the different fields highlighted.
+
+# First we will use the bwa samse command to
+# convert the .sai file to SAM format:
+${bwa} samse data/ref_genome/ecoli_rel606.fasta \
+       results/sai/SRR097977.aligned.sai \
+       data/trimmed_fastq/SRR097977.fastq_trim.fastq > \
+       results/sam/SRR097977.aligned.sam
+
+# Explore the information within your SAM file:
+head results/sam/SRR097977.aligned.sam
+
+# Now convert the SAM file to BAM format for use by downstream tools:
+samtools view -S -b results/sam/SRR097977.aligned.sam > results/bam/SRR097977.aligned.bam
+
+# Sort BAM file by coordinates
+# Sort the BAM file:
+samtools sort results/bam/SRR097977.aligned.bam > results/bam/SRR097977.aligned.sorted.bam
+
+# Variant calling
+# A variant call is a conclusion that there is a nucleotide difference
+# vs. some reference at a given position in an individual genome or
+# transcriptome, often referred to as a Single Nucleotide Polymorphism
+# (SNP). The call is usually accompanied by an estimate of variant
+# frequency and some measure of confidence. Similar to other steps
+# in this workflow, there are number of tools available for variant
+# calling. In this workshop we will be using bcftools, but there are
+# a few things we need to do before actually calling the variants.
+
+# Step 1: Calculate the read coverage of positions in the genome
+# Do the first pass on variant calling by counting read coverage
+# with samtools mpileup:
+samtools mpileup -g -f data/ref_genome/ecoli_rel606.fasta \
+         results/bam/SRR097977.aligned.sorted.bam > results/bcf/SRR097977_raw.bcf
+
+
+# Step 2: Detect the single nucleotide polymorphisms (SNPs)
+# Identify SNPs using bcftools:
+bcftools call --skip-variants indels --multiallelic-caller \
+        --variants-only -O v results/bcf/SRR097977_raw.bcf \
+         -o results/vcf/SRR097977_final_variants.vcf
+
+
+# Assess the alignment (visualization) - optional step
+# In order for us to look at the alignment files in a
+# genome browser, we will need to index the BAM file using samtools:
+samtools index results/bam/SRR097977.aligned.sorted.bam
+
+# SnpSift Concordance
```

Comment:
Should we move this to the validate stage? It's a minor issue but that might help users get a clear map between popper pipeline stages and what's being executed. So there can be another script inside the container that compares vcf files, and then on the `validate.sh` script we invoke docker so that it does the comparison.

Reply:
Yeah, actually I'm going to re-do all the pipeline stages, so each stage calls docker with its script. For example, run.sh will call docker with *docker exec <options> <container> **docker-run.sh***, so each stage is completely isolated from the others, this way will be clearer to understand what each stage is doing. What do you think?